subsys/bluetooth/shell/audio.c

```diff
@@ -1575,3 +1575,78 @@ static int cmd_audio(const struct shell *sh, size_t argc, char **argv)
 
 SHELL_CMD_ARG_REGISTER(audio, &audio_cmds, "Bluetooth audio shell commands",
 		       cmd_audio, 1, 1);
+
+ssize_t audio_ad_data_add(struct bt_data *data, const size_t data_size, const bool discoverable,
+			  const bool connectable)
+{
+	static const uint8_t ad_ext_uuid16[] = {
+		IF_ENABLED(CONFIG_BT_MICP_MIC_DEV, (BT_UUID_16_ENCODE(BT_UUID_MICS_VAL),))
+		IF_ENABLED(CONFIG_BT_ASCS, (BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL),))
+		IF_ENABLED(CONFIG_BT_BASS, (BT_UUID_16_ENCODE(BT_UUID_BASS_VAL),))
+		IF_ENABLED(CONFIG_BT_PACS, (BT_UUID_16_ENCODE(BT_UUID_PACS_VAL),))
+		IF_ENABLED(CONFIG_BT_GTBS, (BT_UUID_16_ENCODE(BT_UUID_GTBS_VAL),))
+		IF_ENABLED(CONFIG_BT_TBS, (BT_UUID_16_ENCODE(BT_UUID_TBS_VAL),))
+		IF_ENABLED(CONFIG_BT_VCS, (BT_UUID_16_ENCODE(BT_UUID_VCS_VAL),))
+		IF_ENABLED(CONFIG_BT_HAS, (BT_UUID_16_ENCODE(BT_UUID_HAS_VAL),))
+	};
+	size_t ad_len = 0;
+
+	if (!discoverable) {
+		return 0;
+	}
+
+	if (ARRAY_SIZE(ad_ext_uuid16) > 0) {
+		__ASSERT(data_size > ad_len, "No space for AD_UUID16");
```

Comment:
Furthermore, the UUIDs of the services is not required (they are a "should"), but the service data (e.g. `ad_cap_announcement` and `ad_bap_announcement`) are "shall"s. 

We should put the service data before the UUIDs so that the UUIDs in the advertising data is not prioritized higher than the service data.

Reply:
> `ad_len` is always 0 here; was this meant to be in a loop?

`ad_len` is 0, but `data_size` is provided by user, which might be 0, so the check is still valid

> We should put the service data before the UUIDs so that the UUIDs in the advertising data is not prioritized higher than the service data.

OK